geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -450,6 +452,8 @@ public void setShutDownAllStatus(int newStatus) {
 
   private final PartitionedRegion colocatedWithRegion;
 
+  private final ColocationLoggerFactory colocationLoggerFactory;
+
   private ColocationLogger missingColocatedRegionLogger;
```

Comment:
It would be more correct if this field were changed to `volatile`

Reply:
I agree.